### PR TITLE
Code randomization

### PR DESCRIPTION
**Pull request by mgramigna**
**Important**: In order to test this PR, you need to add the valuesets to the proper directory. They are .gitignored by default. See the slack channel for more details.

This PR builds off of the prior support added for using ValueSets during patient generation, and selects a random code from a given valueset during each state's `process` step. At a high level, patient generation now works as follows:

1. If any modules specify to use a valueset, load all the valuesets in `src/main/resources/terminology` and place them into one FHIR Bundle to be used for lookup
2. If a valueset is specified for a given state, pick a random code from that valueset when processing that state. This functionality exists in pretty much any state that supports `codes`

---

I have updated the EXM130 R4 module to use valuesets for both the encounter and the procedure as a proof of concept. You can see the functionality by generating some patients and inspecting the output:

```
./run_synthea -p 10 -m exm130-r4 --exporter.years_of_history=0 -a 51-65
```

Some patients should have different codes for their colonoscopy procedure of office visit encounter, since it will pick from the valueset each time that state is processed. I didn't update the other modules, but this functionality should work for all R4 modules we have.

---

One known limitation that I mentioned in the slack:

> there are some states in synthea that specify the “end” of something (e.g. ConditionEnd). When you add these states, in order to specify which condition or whatever it refers to, one option is to add a code. However if during the ConditionOnset state you specified a valueset, there will be no knowledge at the module builder level of what code is picked, so you can’t specify a code to end the condition. I see this being a problem

I think the ability to support specifying a valueset for an end state could be different tasking. 

---

If you read this far, congrats. Have fun testing!

### CODE REVIEW

File: src/main/java/org/mitre/synthea/export/FhirR4.java [hunk 1]
```diff
@@ -1455,7 +1456,8 @@ private static BundleEntryComponent allergy(BundleEntryComponent personEntry, Bu
 
     allergyResource.setPatient(new Reference(personEntry.getFullUrl()));
     Code code = allergy.codes.get(0);
-    allergyResource.setCode(mapCodeToCodeableConcept(code, SNOMED_URI));
+    String system = ExportHelper.getSystemURI(code.system);
+    allergyResource.setCode(mapCodeToCodeableConcept(code, system));
```
Comment (okeefm) [inline src/main/java/org/mitre/synthea/export/FhirR4.java:1460]: These changes away from using only SNOMED/LOINC/whatever system codes for each resource type may be bigger changes than we think when we go to merge this back into upstream Synthea. Their wiki describes each resource/state type as using a particular system or set of systems; I think this change will remove that constraint: https://github.com/synthetichealth/synthea/wiki/Generic-Module-Framework%3A-Basics#codes

**Review by okeefm — APPROVED** [review]
Tested, the valueset randomization code looks good to me. I made one comment about the relaxing of restrictions on codesystems that's probably worth some discussion (in Slack maybe?), but otherwise looks good to me. Also, the tests all pass locally for me.

**Review by okeefm — CHANGES_REQUESTED** [review]
I think we're missing at least one state. `ImagingStudy` has `modality` codes which could technically come from a valueset. You can see an example of a `modality` code in the `congestive_heart_failure` module here: https://github.com/projecttacoma/synthea/blob/abacus/src/main/resources/modules/congestive_heart_failure.json#L425. Here's an example valueset with these modalities in it: http://hl7.org/fhir/STU3/valueset-dicom-cid29.html

**Comment by mgramigna** [comment]
@okeefm Ugh. This feels similar to an issue @hossenlopp brought up when reviewing https://github.com/projecttacoma/module-builder/pull/1.

I think we will need tasking to support nested valueset attributes that are not at the root level of a state, in both the module builder and the engine. I think we are not fully set on the format yet... so I might propose we pawn this off to a new task?

**Review by okeefm — APPROVED** [review]

**Review by jbradl11 — APPROVED** [review]

**Pull request merged**